File: event-schemas/data/csharp/Location.cs
namespace entities
{
  using System.Collections.Generic;

  public class Location
  {
    private string city;
    private string country;
    private string continent;
    private double? latitude;
    private double? longitude;
    private double? speed;
    private Dictionary<string, dynamic> additionalProperties;

    public string City 
    {
      get { return city; }
      set { city = value; }
    }

    public string Country 
    {
      get { return country; }
      set { country = value; }
    }

    public string Continent 
    {
      get { return continent; }
      set { continent = value; }
    }

    public double? Latitude 
    {
      get { return latitude; }
      set { latitude = value; }
    }

    public double? Longitude 
    {
      get { return longitude; }
      set { longitude = value; }
    }

    public double? Speed 
    {
      get { return speed; }
      set { speed = value; }
    }

    public Dictionary<string, dynamic> AdditionalProperties 
    {
      get { return additionalProperties; }
      set { additionalProperties = value; }
    }
  }
}

File: event-schemas/src/entities/Location.ts


class Location {
  private _city?: string;
  private _country?: string;
  private _continent?: string;
  private _latitude?: number;
  private _longitude?: number;
  private _speed?: number;
  private _additionalProperties?: Map<String, object | string | number | Array<unknown> | boolean | null>;

  constructor(input: {
    city?: string,
    country?: string,
    continent?: string,
    latitude?: number,
    longitude?: number,
    speed?: number,
  }) {
    this._city = input.city;
    this._country = input.country;
    this._continent = input.continent;
    this._latitude = input.latitude;
    this._longitude = input.longitude;
    this._speed = input.speed;
  }

  get city(): string | undefined { return this._city; }
  set city(city: string | undefined) { this._city = city; }

  get country(): string | undefined { return this._country; }
  set country(country: string | undefined) { this._country = country; }

  get continent(): string | undefined { return this._continent; }
  set continent(continent: string | undefined) { this._continent = continent; }

  get latitude(): number | undefined { return this._latitude; }
  set latitude(latitude: number | undefined) { this._latitude = latitude; }

  get longitude(): number | undefined { return this._longitude; }
  set longitude(longitude: number | undefined) { this._longitude = longitude; }

  get speed(): number | undefined { return this._speed; }
  set speed(speed: number | undefined) { this._speed = speed; }

  get additionalProperties(): Map<String, object | string | number | Array<unknown> | boolean | null> | undefined { return this._additionalProperties; }
  set additionalProperties(additionalProperties: Map<String, object | string | number | Array<unknown> | boolean | null> | undefined) { this._additionalProperties = additionalProperties; }
}
export default Location;
